pystac/collection.py

```diff
@@ -726,7 +726,7 @@ def get_assets(
             and/or ``role`` if set or else all of this collection's assets.
         """
         if media_type is None and role is None:
-            return dict(self.assets.items())
+            return self.assets
```

Comment:
This feels a little inconsistent to me, since sometimes we get a reference to the actual object (if `media_type` and `role` are `None`) and sometimes we get a wholly new object. We should either return one or the other, and since we can't return a slice view of the interior object (from the filters) we probably have to deepcopy?

Reply:
Yeah that makes sense. I am pretty ambivalent about the usefulness of a method like `get_assets` anyways. It feels more natural to just access the assets directly via the attribute. I guess the point is to make it easier to filter?

Reply:
Yeah I think the filter's the main purpose -- I rarely if every use `get_assets` myself for exactly that reason, and that I don't know if I'm getting a mutable reference or a copy.